lib/pluginmanager/gem_installer.rb

```diff
@@ -0,0 +1,78 @@
+# encoding: utf-8
+require "pluginmanager/ui"
+require "pathname"
+require "rubygems/package"
+
+module LogStash module PluginManager
+  # Install a physical gem package to the appropriate location inside logstash
+  # - Extract the gem
+  # - Generate the specifications
+  # - Copy the data in the right folders
+  class GemInstaller
+    GEM_HOME = Pathname.new(::File.join(LogStash::Environment::BUNDLE_DIR, "jruby", "1.9"))
+    SPECIFICATIONS_DIR = "specifications"
+    GEMS_DIR = "gems"
+
+    attr_reader :gem_home
+
+    def initialize(gem_file, display_post_install_message = false, gem_home = GEM_HOME)
```

Comment:
> do we ever want to not show the post install message?
  
Yes we do, we only really want the `post_install_message` for plugins that will be added to the gemfile (our gems). But all gems are free to have their own message but it doesn't mean we want to display the output to the users.